src/main/java/teammates/storage/entity/StudentProfile.java

```diff
@@ -18,6 +19,11 @@
 @PersistenceCapable
 public class StudentProfile {
 
+    // CHECKSTYLE.OFF:JavadocVariable self-documenting
+    @NotPersistent
+    public static final String PRIMARY_KEY_NAME = Entity.getFieldWithPrimaryKeyAnnotation(StudentProfile.class);
```

Comment:
Might be better to use a method instead of a field:
```java
public static String getPrimaryKeyName() {
    return Entity....;
}
```
Reasons being getters are exempted from Javadoc (for now, perhaps forever?) and you will not need the `@NotPersistent` annotation anymore.